plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/XmlTest.java

```diff
@@ -0,0 +1,53 @@
+package com.google.cloud.tools.eclipse.newproject.whitelist;
+
+import java.io.File;
+import java.io.IOException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.eclipse.core.runtime.IExtensionPoint;
+import org.eclipse.core.runtime.IExtensionRegistry;
+import org.eclipse.core.runtime.RegistryFactory;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+// todo: extract common code into a utility class
+public class XmlTest {
+    
+  private Document document;
+	
+  @Before
+  public void setUp() throws ParserConfigurationException, IOException, SAXException {
+    DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+    factory.setNamespaceAware(true);
+    DocumentBuilder builder = factory.newDocumentBuilder();
+    document = builder.parse(
```

Comment:
It would be nice to avoid embedding relative pathnames, especially if we decide to move the tests out to a different root in the repository (e.g., `plugins/`, `features/`, and `tests/`).  But that can wait until we refactor these with a utility class.

But if you want to, you can resolve this file using OSGi (which is necessary to use the extension registry as per below) by fetch the file from the bundle.  Normally finding a bundle would require iterating through the loaded bundles looking at them by name, but since these tests are in a fragment then we share the same bundle.

```
Bundle bundleUnderTest = FrameworkUtil.getBundle(this.getClass());
URL pluginXml = bundleUnderTest.getEntry("plugin.xml");
try(InputStream input = pluginXml.openStream()) {
  document = builder.parse(input);
}
```